cargo-cyclonedx/src/generator.rs

```diff
@@ -374,23 +375,44 @@ pub enum GeneratorError {
 fn top_level_dependencies(
     members: &[Package],
     package_ids: &PackageSet<'_>,
+    resolve: &Resolve,
 ) -> Result<BTreeSet<Package>, GeneratorError> {
     log::trace!("Adding top-level dependencies to SBOM");
     let mut dependencies = BTreeSet::new();
 
     let all_dependencies = members
         .iter()
-        .flat_map(|m| m.dependencies().iter())
-        .filter(|d| d.kind() == DepKind::Normal);
+        .flat_map(|m| {
+            resolve
+                .deps(m.package_id())
+                .filter_map(move |r| match r.0 == m.package_id() {
+                    true => Some(r.1),
+                    false => None,
+                })
```

Comment:
Style preference: In my opinion, this is cleaner as the following and likely can be optimized to similar instructions.

```suggestion
                .filter(move |r| r.0 == m.package_id())
                .map(|(_, dependency)| dependency)
```

Not required, but thought it was worth suggesting

Reply:
Works for me, I don't have any strong opinion there.